[VP][RISCV] Add vp.reduce.fmaximum/fminimum and its RISC-V codegen

mshockwave opened the pull request:
`vp.reduce.fmaximum/fminimum` are the VP version of `vector.reduce.fmaximum/fminimum`.

mshockwave committed:
[VP][RISCV] Add vp.reduce.fmaximum/fminimum and its RISC-V codegen

`vp.reduce.fmaximum/fminimum` are the VP version of
`vector.reduce.fmaximum/fminimum`.

topperc commented on llvm/test/CodeGen/RISCV/rvv/vp-reduce.ll:
```diff
@@ -0,0 +1,205 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 4
```
This is an overly generic test name. How are the other vp.reduce tests structured?

mshockwave replied:
I'd consolidates the tests with other floating point vp.reduce tests.

mshockwave committed:
Address review comments

 - Check the start value of vp.reduce for NaN as well.
 - Consolidate the tests with other vp.reduce.* test.

mshockwave commented on llvm/lib/Target/RISCV/RISCVISelLowering.cpp:
```diff
@@ -9804,8 +9811,25 @@ SDValue RISCVTargetLowering::lowerVPREDUCE(SDValue Op,
+      Op->getFlags().hasNoNaNs())
+    return Res;
+
+  // Propagate NaNs.
```
Good point, it's fixed now.

topperc reviewed (approved):
LGTM

mshockwave committed:
Fix LangRef.rst build error
